// vite-preset-react/dist/index.cjs
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};

// src/index.ts
__export(exports, {
  default: () => reactPlugin
});

// node_modules/.pnpm/tsup@5.4.0_typescript@4.4.3/node_modules/tsup/assets/cjs_shims.js
var importMetaUrlShim = typeof document === "undefined" ? new (require("url")).URL("file:" + __filename).href : document.currentScript && document.currentScript.src || new URL("main.js", document.baseURI).href;

// src/index.ts
var import_plugin_react = __toModule(require("@vitejs/plugin-react"));

// src/devtools.ts
function reactDevtoolsPlugin({
  removeInProd = false
} = {}) {
  const plugin = {
    name: "react:devtools",
    enforce: "pre",
    apply: "build",
    transformIndexHtml(code) {
      if (removeInProd) {
        return {
          html: code,
          tags: [
            {
              injectTo: "body",
              tag: `script`,
              children: `if (typeof window.__REACT_DEVTOOLS_GLOBAL_HOOK__ === 'object') { window.__REACT_DEVTOOLS_GLOBAL_HOOK__.inject = function () {};};`
            }
          ]
        };
      }
      return code;
    }
  };
  return plugin;
}

// src/index.ts
function reactPlugin({
  removeDevtoolsInProd = false,
  injectReact = true,
  reactPluginOptions: reactRefreshOptions
} = {}) {
  return [
    (0, import_plugin_react.default)(__spreadValues({ jsxRuntime: injectReact ? "automatic" : "classic" }, reactRefreshOptions)),
    reactDevtoolsPlugin({ removeInProd: removeDevtoolsInProd })
  ];
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {});
